CMakeLists.txt

```
@@ -36,7 +44,16 @@ endif()
 
 add_compile_definitions(
   $<$<PLATFORM_ID:Windows>:WIN>
-  $<$<PLATFORM_ID:Windows>:IN_80386>)
+  $<$<CONFIG:Debug>:DEBUG>
+)
+
+# Enable architecture-specific optimizations
+if("${CMAKE_SIZEOF_VOID_P}" EQUAL "4")
+    add_compile_definitions($<$<PLATFORM_ID:Windows>:IN_80386>)
+else()
+    # FUTURE: Remove when we have 3DMM compiling for x64
+    message(FATAL_ERROR "Cannot compile for 64-bit yet")
+endif()
```

CMake comparisons don't require strings in all cases. Additionally, just adding the `IN_80386` earlier is fine as things will stop if you use the following instead.

```suggestion
if (NOT CMAKE_SIZEOF_VOID_P EQUAL 4)
  message(FATAL_ERROR "Cannot compile for 64-bit yet")
endif()
```